File: android/java/v1/AmpliApp/app/src/main/java/com/example/ampliapp/LoggingMiddleware.java
package com.example.ampliapp;

import com.amplitude.api.*;

import org.json.JSONException;

public class LoggingMiddleware implements Middleware {
    @Override
    public void run(MiddlewarePayload payload, MiddlewareNext next) {
        String event = null;
        if (payload.event != null) {
            try {
                event = payload.event.toString(2);
            } catch (JSONException e) {
                System.err.println(e.toString());
            }
        }

        String extra = null;
        if (payload.extra != null) {
            extra = payload.extra.toString();
        }

        System.out.printf("[ampli] event=%s, extra=%s%n", event, extra);

        next.run(payload);
    }
}

File: android/kotlin/v1/AmpliApp/app/src/main/java/com/example/ampliapp/LoggingMiddleware.kt
package com.example.ampliapp

import com.amplitude.api.*

class LoggingMiddleware : Middleware {
    override fun run(payload: MiddlewarePayload?, next: MiddlewareNext?) {
        println("[ampli] event=${payload?.event?.toString(2)}, extra=${payload?.extra}")

        next?.run(payload)
    }
}